# HG changeset patch
# User Richard Marti <richard.marti@gmail.com>
# Date 1787140545 -7200
# Parent  8e35f70342f8c5b6bf0be8c02cfd41d4210b6672
Bug 1842493 - Move the menubar on top of the toolbars.

Contains hunks of 10-misc-unified-toolbar-after-menu.patch

diff --git a/mail/base/content/messenger.xhtml b/mail/base/content/messenger.xhtml
--- a/mail/base/content/messenger.xhtml
+++ b/mail/base/content/messenger.xhtml
@@ -495,9 +495,8 @@
 -->
 <toolbox id="navigation-toolbox" labelalign="end" defaultlabelalign="end">
 
   <vbox id="titlebar">
-    <html:unified-toolbar></html:unified-toolbar>
     <!-- Menu -->
     <toolbar id="toolbar-menubar"
              class="chromeclass-menubar themeable-full"
              type="menubar"
@@ -512,9 +511,12 @@
              prependmenuitem="true">
 # The entire main menubar is placed into messenger-menubar.inc.xhtml, so that it
 # can be shared with other top level windows.
 #include messenger-menubar.inc.xhtml
+<spacer flex="1" skipintoolbarset="true"/>
+#include messenger-titlebar-items.inc.xhtml
     </toolbar>
+    <html:unified-toolbar></html:unified-toolbar>
 
     <toolbar id="tabs-toolbar" class="chromeclass-toolbar" collapsed="true">
       <tabs is="tabmail-tabs" id="tabmail-tabs"
             flex="1"
diff --git a/mail/themes/shared/mail/messenger.css b/mail/themes/shared/mail/messenger.css
--- a/mail/themes/shared/mail/messenger.css
+++ b/mail/themes/shared/mail/messenger.css
@@ -71,12 +71,20 @@ hr {
   border-block-start: 1px solid var(--field-border-color);
   border-block-end: 0;
 }
 
-:root:not([customtitlebar]) .titlebar-buttonbox-container {
+/* Hide the Unified toolbar controls if the menubar is permanently shown.
+ * (That is, if the menu bar doesn't autohide, and we're not in a fullscreen or
+ * popup window.) */
+:root:not([customtitlebar]) .titlebar-buttonbox-container,
+:root[customtitlebar] #toolbar-menubar:not([autohide="true"]) + unified-toolbar > div > .titlebar-buttonbox-container {
   visibility: collapse;
 }
 
+:root[customtitlebar] #toolbar-menubar[autohide="true"]:not([inactive="true"], [customizing="true"]) + unified-toolbar > div > .titlebar-buttonbox-container {
+  visibility: hidden;
+}
+
 :root:not([sizemode="maximized"]) .titlebar-restore,
 :root:is([sizemode="maximized"]) .titlebar-max {
   display: none;
 }
@@ -126,9 +134,9 @@ hr {
 :root[lwtheme] #compose-toolbox:-moz-window-inactive {
   background-color: var(--lwt-accent-color-inactive, var(--lwt-accent-color));
 }
 
-#titlebar > #toolbar-menubar:-moz-window-inactive,
+#titlebar > #toolbar-menubar > #mail-menubar > menu:-moz-window-inactive,
 #titlebar > #tabs-toolbar:-moz-window-inactive {
   color: color-mix(in srgb, currentColor 70%, transparent);
 }
 
diff --git a/mail/themes/shared/mail/unifiedToolbar.css b/mail/themes/shared/mail/unifiedToolbar.css
--- a/mail/themes/shared/mail/unifiedToolbar.css
+++ b/mail/themes/shared/mail/unifiedToolbar.css
@@ -144,8 +144,12 @@ unified-toolbar-customization {
     padding: var(--unified-customization-padding);
   }
 }
 
+.customizingUnifiedToolbar[tabsintitlebar] #titlebar > unified-toolbar > #unifiedToolbarContainer > .titlebar-buttonbox-container {
+  visibility: visible;
+}
+
 #customizationHeading {
   align-self: center;
   font-size: 1.2rem;
   font-weight: 600;
